### DISCUSSION TITLE
Best Practive To Improve PageSpeed

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3888] [1. Help] [answered]
When running [https://www.swisswebcams.ch/webcam/1511343694 (as an example) on ](https://pagespeed.web.dev/analysis/https-rc-swisswebcams-ch-webcam-1511343694/8v61s1j13f?form_factor=mobile) as an example, a big issue is `Script Evaluation` with 14 seconds.

![image](https://github.com/alpinejs/alpine/assets/1435963/1cbb4e4a-4923-43ae-9aff-ac8a7c18d6cc)

**My Thoughts**
1. I'm pretty sure the lighthouse simulation is not optimized for AlpineJS because I can't hardly see any reason this can take _that_ long
2. However, I would like to know/discuss the best practices to optimze this measurement (code maintainability/readability vs speed)
3. Perhaps, during the build process there could be used/implemented some optimizations (for example using vite)?

I'm interested to read about different opinions on this.

**Comment by ekwoka** [comment, marked as answer]
Well, the answer is that "this isn't alpine".

You have about over 2MB of transferred JS.

Half your execution time is `lighthouse-eval.js` which seems to be some kind of leak in lighthouse that has only started happening recently as far as I can tell. 

Your actual alpine code is only 38kb, and seems to have an execution of 2-3.5s.

You do have quite a lot of stuff that Alpine needs to do at initialization (especially with the x-data on the document element). If you want more info, you'd probably need to disable all the 2mb of js except alpine stuff and run the profiler and see where it's spending it's time.

**Reply by simonferndriger** [reply]
You are right, there is definitely something weird going on there. I filed a bug report for pagespeed. Thanks!
